src/main/java/org/jenkinsci/main/modules/instance_identity/InstanceIdentity.java

```diff
@@ -53,19 +53,25 @@ public InstanceIdentity(File keyFile) throws IOException {
             keys = read(new FileReader(oldKeyFile), gen);
             write(keys, keyFile);
             Util.deleteFile(oldKeyFile);
-        } else if (keyFile.exists()) {
+        } else {
             byte[] enc = FileUtils.readFileToByteArray(keyFile);
```

Comment:
Well now this is no good, since if `!keyFile.exists()` then this line is just going to throw an exception.

It is just a waste of time to edit code like this without test coverage demonstrating the problem and proving your fix. There are some cases where creating a test for an area of code is really hard for technical reasons, but I doubt this is one of them.